specs/multicloud/Huawei_Cloud_File_Block.md

```diff
@@ -0,0 +1,931 @@
+                "Kind": {
+                    "StringValue": "SFS"
+                }
+            },
+            "IsPublic": {
+                "Kind": {
+                    "BoolValue": false
+                }
+            },
+            "Links": {
+                "Kind": {
+                    "StructValue": {
+                        "fields": {
+                            "href": {
+                                "Kind": {
+                                    "StringValue": "<Links>"
+                                }
+                            },
+                            "rel": {
+                                "Kind": {
+                                    "StringValue": "self"
+                                }
+                            }
+                        }
+                    }
+                }
+            },
+            "ProjectID": {
+                "Kind": {
+                    "StringValue": "<ProjectID>"
+                }
+            },
+            "ShareID": {
+                "Kind": {
+                    "StringValue": "cf149d37-09f6-45eb-8c73-8713f0abffeb"
+                }
+            },
+            "ShareNetworkID": {
+                "Kind": {
+                    "StringValue": ""
+                }
+            },
+            "ShareType": {
+                "Kind": {
+                    "StringValue": "default"
+                }
+            },
+            "SnapshotID": {
+                "Kind": {
+                    "StringValue": ""
+                }
+            },
+            "VolumeType": {
+                "Kind": {
+                    "StringValue": "default"
+                }
+            }
+        }
+    }
+}
+```
+**List File shares**
+
+GET http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares
+
+Response:
+
+```
+{
+    "fileshares": [
+        {
+            "id": "5f87b9bc90f7e10001fa260c",
+            "createdAt": "2020-10-15T08:23:48",
+            "updatedAt": "2020-10-15T08:23:57",
+            "name": "Newfs",
+            "description": "Huawei SFS FileShare",
+            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
+            "userId": "558057c4256545bd8a307c37464003c9",
+            "backendId": "5f87b48ea35cea0001db816d",
+            "backend": "hw-backend-file",
+            "size": 1073741824,
+            "type": "cloudFS",
+            "region": "cn-north-1",
+            "availabilityZone": "cn-north-1b",
+            "status": "available",
+            "protocols": [
+                "NFS"
+            ],
+            "tags": [
+                {
+                    "key": "Name",
+                    "value": "Newfs"
+                }
+            ],
+            "encrypted": true,
+            "encryptionSettings": {
+                "DomainId": "<DomainId>",
+                "KmsKeyId": "<KmsKeyId>",
+                "KmsKeyName": "<KmsKeyName>"
+            },
+            "metadata": {
```

Comment:
how did we arrive at this JSON response ? do we pass-through the response from the vendor API or this is the SODA response format, common across any vendor ?

Reply:
This is SODA response format, consolidated across providers